subsys/net/lib/mcumgr_smp_client/src/mcumgr_smp_client_shell.c

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <ctype.h>
+#include <zephyr/drivers/gpio.h>
+#include <stdio.h>
+#include <app_event_manager.h>
+#include <date_time.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MCUmgr_smp_client_shell, CONFIG_NRF_MCUMGR_SMP_CLIENT_LOG_LEVEL);
+
+#include <mcumgr_smp_client.h>
+#include <zephyr/shell/shell.h>
+
+static bool active_download;
+
+static int hash_to_string(char *hash_string, size_t string_size, uint8_t *hash)
+{
+	char *ptr = hash_string;
+	int buf_size = string_size;
+	int len = 0;
+
+	for (int i = 0; i < 32; i++) {
+		len += snprintk(ptr + len, buf_size - len, "%x", hash[i]);
+		if (len >= string_size) {
+			return -1;
+		}
+	}
+	hash_string[len] = 0;
+
+	return 0;
+}
+
+static void fota_download_shell_callback(const struct fota_download_evt *evt)
+{
+	switch (evt->id) {
+	/* These two cases return immediately */
+	case FOTA_DOWNLOAD_EVT_PROGRESS:
+		return;
+	default:
+		return;
+
+	/* Following cases mark end of FOTA download */
+	case FOTA_DOWNLOAD_EVT_CANCELLED:
+		LOG_ERR("FOTA_DOWNLOAD_EVT_CANCELLED");
+		active_download = false;
+		break;
+	case FOTA_DOWNLOAD_EVT_ERROR:
+		LOG_ERR("FOTA_DOWNLOAD_EVT_ERROR: %d", evt->cause);
+		active_download = false;
+		break;
+
+	case FOTA_DOWNLOAD_EVT_FINISHED:
+		LOG_INF("FOTA download finished");
+		active_download = false;
+		break;
+	}
+}
+
+static void print_image_list(const struct shell *sh, struct mcumgr_image_state *image_list)
+{
+	struct mcumgr_image_data *list;
+	char hash_string[(IMG_MGMT_HASH_LEN * 2) + 1];
+
+	list = image_list->image_list;
+	for (int i = 0; i < image_list->image_list_length; i++) {
+		if (list->flags.active) {
+			shell_print(sh, "Primary Image(%d) slot(%d)", list->img_num,
+				    list->slot_num);
+		} else {
+			shell_print(sh, "Secondary Image(%d) slot(%d)", list->img_num,
+				    list->slot_num);
+		}
+
+		shell_print(sh, "       Version: %s", list->version);
+		shell_print(sh, "       Bootable(%d) Pending(%d) Confirmed(%d)",
+			    list->flags.bootable, list->flags.pending, list->flags.confirmed);
+		if (hash_to_string(hash_string, sizeof(hash_string), list->hash) == 0) {
+			shell_print(sh, "       Hash: %s", hash_string);
```

Comment:
Any reason not to use \t for spacing?

Reply:
I have clone mcuboot command outout that output looks similar. That is only reason